### PR TITLE
UCT/UD: Fix out-of-order with pending queue

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8123/ for details (Mellanox internal link).

File: src/uct/ib/ud/base/ud_ep.c [hunk 1]
```diff
@@ -692,6 +692,7 @@ void uct_ud_ep_process_rx(uct_ud_iface_t *iface, uct_ud_neth_t *neth, unsigned b
         goto out;
     }
 
+        uct_ud_iface_raise_pending_async_ev(iface);
```
Comment (brminich) [inline src/uct/ib/ud/base/ud_ep.c]: indentation

File: src/uct/ib/ud/base/ud_ep.c [hunk 2]
```diff
@@ -1147,9 +1144,18 @@ uct_ud_ep_do_pending(ucs_arbiter_t *arbiter, ucs_arbiter_elem_t *elem,
         }
         return UCS_ARBITER_CB_RESULT_REMOVE_ELEM;
     }
+
     /* try to send ctl messages */
     uct_ud_ep_do_pending_ctl(ep, iface);
-    return uct_ud_ep_ctl_op_next(ep);
+    if (in_async_progress) {
+        return uct_ud_ep_ctl_op_next(ep);
+    } else {
```
Comment (brminich) [inline src/uct/ib/ud/base/ud_ep.c]: why do you need this `if else`? If there is any control operation `uct_ud_ep_ctl_op_next` would return UCS_ARBITER_CB_RESULT_NEXT_GROUP anyway.

File: src/uct/ib/ud/base/ud_ep.c [hunk 3]
```diff
@@ -701,7 +702,6 @@ void uct_ud_ep_process_rx(uct_ud_iface_t *iface, uct_ud_neth_t *neth, unsigned b
          * initiates sends from any endpoint created on the iface.
          * This flag would be cleared after all incoming messages
```
Comment (evgeny-leksikov) [inline src/uct/ib/ud/base/ud_ep.c]: Seems like this comment is not relevant

File: src/uct/ib/ud/base/ud_ep.c (hunk 2 above)
Comment (brminich) [reply src/uct/ib/ud/base/ud_ep.c]: aa, the ctl ops might have been cleared during `uct_ud_ep_do_pending_ctl` handling, right?

Comment (yosefe) [reply src/uct/ib/ud/base/ud_ep.c]: yes, will add comment

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8130/ for details (Mellanox internal link).

**Pull request merged**